### ISSUE TITLE
improve previous line count

### ISSUE THREAD

**Issue by SamVerschueren** [sindresorhus/log-update#12]
The previous line count is being tracked by `log-update` but it does not take the terminal size into account.

For instance, if you log a line of 40 characters while the terminal is only 30 characters, it will be printed on 2 lines. After calling `log-update` again, it will only erase the 2nd line because the previous line count was only 1 and not 2.

So we have to iterate over all the lines [here](https://github.com/sindresorhus/log-update/blob/master/index.js#L12) and calculate the number of printed lines instead.

**Comment by sindresorhus** [comment]
When I made `log-update` I thought the terminal handled that, but I guess that would be too easy...

**Comment by kevva** [comment]
Wouldn't something like this work?

```js
prevLineCount = out.match(new RegExp(`[^]{1,${columns}}`, 'g')).join('\n').split('\n').length;
```

**Comment by sindresorhus** [comment]
That doesn't handle ansi escapes... I think we'll have to use [`wrap-ansi`](https://github.com/chalk/wrap-ansi).

**Comment by SamVerschueren** [comment]
@kevva Are you looking into this? If not, happy to take this one. Just let me know :).

**Comment by kevva** [comment]
Feel free if you have time :).

**Comment by SamVerschueren** [comment]
Will have to drop support for Node 0.10 and 0.12 because `term-size` does not support them. Feedback?

**Comment by sindresorhus** [comment]
That's fine. I'm doing that with all my modules now when needed.

**Comment by Jameskmonger** [comment]
I can confirm that this is still an issue.

```
const logUpdate = require("log-update");

let first = `It was getting dark, and we weren’t there yet.
Italy is my favorite country; in fact, I plan to spend two weeks there next year.
Cats are good pets, for they are clean and are not noisy.
Where do random thoughts come from?
Christmas is coming.`.replace(/\s/g,'')

let second = `The river stole the gods.
We need to rent a room for our party.
The lake is a long way from here.
Is it free?
We have never been to Asia, nor have we visited Africa.`.replace(/\s/g,'')

let third = `Mary plays the piano.
The shooter says goodbye to his love.
The lake is a long way from here.
Should we start class now, or should we wait for everyone to get here?
We have a lot of rain in June.`.replace(/\s/g,'')

logUpdate(first);

setTimeout(() => logUpdate(second), 1000);

setTimeout(() => logUpdate(third), 2000);
```

Final output:

```
λ node index.js
Itwasgettingdark,andweweren’tthereyet.Italyismyfavoritecountry;infact,Iplantospendtwoweekstherenextyear.Catsaregoodpets,Theriverstolethegods.Weneedtorentaroomforourparty.Thelakeisalongwayfromhere.Isitfree?WehaveneverbeentoAsia,norhavewevisiMaryplaysthepiano.Theshootersaysgoodbyetohislove.Thelakeisalongwayfromhere.Shouldwestartclassnow,orshouldwewaitforeveryonetogethere?WehavealotofraininJune.
```

**Comment by Jameskmonger** [comment]
I should have clarified that the above occurs both before and after @SamVerschueren 's fix.

**Comment by Jameskmonger** [comment]
This can be fixed by checking for overflows like so:

```
let lines = out.split('\n');
prevLineCount = lines.length;

lines.forEach(l => {
	prevLineCount += Math.floor(l.length, process.stdout.columns);
});
```

**Comment by SamVerschueren** [comment]
I will have a look. What terminal are you using? Which OS?

**Comment by Jameskmonger** [comment]
Using cmder on Windows 10 @SamVerschueren.

**Comment by SamVerschueren** [comment]
Thanks, will look into it.

**Comment by SamVerschueren** [comment]
@Jameskmonger Have you tested it with PR #16?

**Comment by Jameskmonger** [comment]
@SamVerschueren yes I have tested it but it does not fix the issue, testing #16 gives the following final output in cmder in Windows 10:

![cmder windows 10](http://i.imgur.com/TKXhGan.png)

And in command prompt in windows 10:

![command prompt windows 10](http://i.imgur.com/CA6ndeO.png)

---

My fix above gives the following output in cmder in Windows 10:

![cmder windows 10](http://i.imgur.com/lBeYt6r.png)

And in command prompt in windows 10:

![command prompt windows 10](http://i.imgur.com/bgqiIwo.png)
